modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/TxManagerImpl.java

```diff
@@ -188,6 +191,36 @@ public TxStateMeta stateMeta(UUID txId) {
         return txStateMap.get(txId);
     }
 
+    @Override
+    public CompletableFuture<TransactionMeta> transactionMetaReadTimestampAware(
+            UUID txId,
+            HybridTimestamp readTimestamp,
+            TxStateStorage storage
+    ) {
+        AtomicReference<CompletableFuture<TransactionMeta>> txStateFutRef = new AtomicReference<>();
+
+        txStateMap.compute(txId, (k, v) -> {
+            TransactionMeta txMeta = v;
+
+            if (txMeta == null) {
+                txMeta = storage.get(txId);
+            }
+
+            if (txMeta != null && txMeta instanceof TxStateMetaFinishing) {
+                txStateFutRef.set(((TxStateMetaFinishing) txMeta).txFinishFuture());
+            } else {
+                // All future transactions will be committed after the resolution processed.
+                clock.update(readTimestamp);
```

Comment:
It a commit partition does not ever generate commit timestamp now, there is no point to update clock on the commit partition.

Reply:
seems we no longer need this entire method